neo/Prompt/Commands/tests/test_send_command.py

```diff
@@ -182,21 +299,219 @@ def test_13_utilst_bad_type(self):
             self.assertEqual(len(args), 0)
             self.assertEqual(len(txattr), 0)
 
-    @patch.object(Send, 'gather_signatures')
-    def test_14_owners(self, mock):
+    def test_fails_to_sign_tx(self):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=[UserWalletTestCase.wallet_1_pass()]):
+            with patch('neo.Wallets.Wallet.Wallet.Sign', return_value=False):
+
+                wallet = self.GetWallet1(recreate=True)
+                args = ['gas', self.watch_addr_str, '2']
+
+                framework = construct_send_basic(wallet, args)
+                res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
+
+                self.assertFalse(res)
+
+    def test_fails_to_relay_tx(self):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=[UserWalletTestCase.wallet_1_pass()]):
+            with patch('neo.Prompt.Commands.Send.NodeLeader.Relay', return_value=False):
+
+                wallet = self.GetWallet1(recreate=True)
+                args = ['gas', self.watch_addr_str, '2']
+
+                framework = construct_send_basic(wallet, args)
+                res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], fee=framework[2], owners=framework[3], user_tx_attributes=framework[4])
+
+                self.assertFalse(res)
+
+    def test_could_not_send(self):
+
         wallet = self.GetWallet1(recreate=True)
+        args = ['gas', self.watch_addr_str, '2']
+
+        contract_tx, scripthash_from, fee, owners, user_tx_attributes = construct_send_basic(wallet, args)
+        res = process_transaction(wallet, contract_tx, scripthash_from, fee, owners, user_tx_attributes)  # forces the 'try:' to fail
+
+        self.assertFalse(res)
+
+    def test_sendmany_good_simple(self):
+        with patch('neo.Prompt.Commands.Send.prompt', side_effect=["neo", self.watch_addr_str, "1", "gas", self.watch_addr_str, "1", UserWalletTestCase.wallet_1_pass()]):
+
+            wallet = self.GetWallet1(recreate=True)
+            args = ["2"]
+
+            framework = construct_send_many(wallet, args)
+            res = process_transaction(wallet, contract_tx=framework[0], scripthash_from=framework[1], scripthash_change=framework[2], fee=framework[3], owners=framework[4], user_tx_attributes=framework[5])
+
+            self.assertTrue(res)  # verify successful tx
+
+            json_res = res.ToJson()
+            self.assertEqual(self.watch_addr_str, json_res['vout'][0]['address'])  # verify correct address_to
```

Comment:
Is it just me going crazy or have i seen this before somewhere? It feels like I'm reviewing something I've done before in another PR. Note this is not a request to change something, I think I just seem to have lost track

Reply:
Probably. I derived these tests from my learning experience from implementing `sendmany` for JsonRpcApi